tensorflow/compiler/xla/service/algebraic_simplifier.cc

```diff
@@ -1581,6 +1581,15 @@ AlgebraicSimplifierVisitor::OptimizeDotOfReorderContractingDims(
       lhs_contracting_dims.Add(i);
     }
   }
+  // We require the "unsquished" lhs contracting dims to be consecutive.
+  auto is_iota = [](absl::Span<const int64> dims) {
```

Comment:
I believe this is `IsIdentityPermutation` in util.cc

Reply:
They are not exactly the same though. IsIdentityPermutation is is_iota plus the first element is 0. But we don't require the first element to be 0 here.

Reply:
Or I can do something like this
`std::vector<int64> temp_lhs_contracting_dims;
for (auto dim : lhs_contracting_dims) {
  temp_lhs_contracting_dims.push_bakc(dim - lhs_contracting_dims[0])
}
if (!IsIdentityPermutation(temp_lhs_contracting_dims) {
  return nullptr;
}`

Reply:
Oh, my bad!  This is fine.